sentry/src/main/java/io/sentry/SentryEnvelopeItem.java

```diff
@@ -86,24 +64,37 @@
     }
   }
 
-  public static @NotNull SentryEnvelopeItem fromEvent(
-      final @NotNull ISerializer serializer, final @NotNull SentryEvent event) throws IOException {
-    Objects.requireNonNull(serializer, "ISerializer is required.");
-    Objects.requireNonNull(event, "SentryEvent is required.");
+  public @Nullable SentryTransaction getTransaction(final @NotNull ISerializer serializer)
+      throws Exception {
+    if (header == null || header.getType() != SentryItemType.Transaction) {
+      return null;
+    }
+    try (final Reader eventReader =
+        new BufferedReader(new InputStreamReader(new ByteArrayInputStream(getData()), UTF_8))) {
+      return serializer.deserializeTransaction(eventReader);
+    }
+  }
+
+  public static @NotNull SentryEnvelopeItem from(
```

Comment:
M: I think this API is a step back. What if `Object` is now a `byte[]`? It seems like an over generalization because now we can pass infinite invalid values. Below is hardcoded `application/json` but what if I'm trying to create an attachment? Or just a `text/plain` and passed a `String`? Is that valid?

Reply:
I brought back an old API.